app/controllers/verify/finance_controller.rb

```diff
@@ -17,20 +19,39 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_finance_form: idv_finance_form,
+        idv_session: idv_session
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
+
     def render_form
-      if finance_params[:finance_type] == 'ccn'
+      if step_params[:finance_type] == 'ccn'
         render :new
       else
         render 'verify/finance_other/new'
       end
     end
 
-    def idv_finance_form
-      @_idv_finance_form ||= Idv::FinanceForm.new(idv_session.params)
+    def track_idv_event
+      result = {
+        success: idv_session.financials_confirmation.success?
+      }
+
+      analytics.track_event(Analytics::IDV_FINANCE_CONFIRMATION, result)
```

Comment:
Thanks @monfresh for clarifying your goals: better analytics coverage is a great idea.

I think we can achieve the same thing w/o overloading the Form class. Maybe this is a naming thing, but to me, a Form represents parameters from an HTTP request, and should be confined to validating those parameters on their own merits. The vendor confirmation of those parameters is a qualitatively different exercise, because (a) it applies logic over which we have no control and (b) goes beyond simply validating the attributes to confirming they signify to a real human being. Together, the Form validation and the vendor confirmation comprise a single Step in the verification process. Having the Form do both things (validate and confirm) co-mingles the concerns, IMO.

The Step also handles the mutation of the `idv_session` which IMO is outside the scope of Form validation.

I'm going to refactor the Step classes slightly to capture the errors more cleanly, and move the Analytics object into the Step so that we log IdV events all in one place.